File: JNibbles/src/main/java/org/lagash/nibbles/Position.java
/**
 * Copyright (C) 2017 DiegoG.
 */
package org.lagash.nibbles;

public class Position implements IPosition {
    private int x, y;

    public int getX() {
        return x;
    }
    
    public void setX(int value) {
        this.x = value;
    }

    public int getY() {
        return y;
    }
    
    public void setY(int value) {
        this.y = value;
    }
    public void setPosition(IPosition position)
    {
        setX( position.getX() );
        setY( position.getY() );
    }

    public boolean isValid()
    {
        return (getX() >= 0) && (getY() >= 0);
    }

    public boolean isValid(int topX, int topY)
    {
        return isValid() && (getX() < topX) && (getY() < topY);
    }

    public void printPosition() {
        System.out.print("x:" + getX());
        System.out.println(" y:" + getY());
    }
}

File: old/Nibbles/Position.cs
namespace Nibbles
{
    class Position : IPosition
    {
        public int X { get; set; }
        
        public int Y { get; set; }

        public void SetPosition(IPosition position)
        {
            X = position.X;
            Y = position.Y;
        }

        public bool IsValid()
        {
            return (X >= 0) && (Y >= 0);
        }

        public bool IsValid(int topX, int topY)
        {
            return IsValid() && (X <= topX) && (Y <= topY);
        }
    }
}